refactor(components): Refactoring, add pre-infinite scroll functional

untael opened the pull request.

untael committed:
refactor(components): Refactoring, add pre-infinite scroll functional

iTechJ commented on .idea/frontend.iml:
```diff
@@ -0,0 +1,12 @@
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
```
В этом проекте смотри сам, но в коммерческом проекте лучше не комитить иде специфичные файлы, как из папки .idea. Можно комитить те которые понимают все ide типо .editorconfig

iTechJ commented on src/Components/ContactExplorer/ContactExplorer.js:
```diff
@@ -132,32 +123,44 @@ class ContactExplorer extends React.Component {
+  getContacts = () => {
+    const pageNumber = this.state.pageNumber
+    console.log(pageNumber)
+    axios.post('http://localhost:3000/list', { pageNumber })
```
Ещё из рекомендаций: добавить в архитектуру сервисы. То есть отдельный класс/функция которую ты импортишь в компонент и вызываешь например `contact.getList(page)`. Таким образом реализация сервиса может поменяться и это не затронет компонент. Ну и при использовании редакса здесь будет диспатч экшена, а в эшене будет вызов сервиса.

untael committed:
refactor (components): Huge rework styles, reuse create component

Did styles for "mobile first" pattern, did reuse ContactCreateForm instead ContactEditForm, remove some useless code
